From 8124584d40a3df72f47d3af4cdfde800a901617f Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Thu, 22 Jan 2026 17:42:51 +0100
Subject: [PATCH] Adapt to Poppler 26.02 API changes

---
 src/lib/pdf/pdfdocument.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/lib/pdf/pdfdocument.cpp b/src/lib/pdf/pdfdocument.cpp
index cec27983..66ffcf6a 100644
--- a/src/lib/pdf/pdfdocument.cpp
+++ b/src/lib/pdf/pdfdocument.cpp
@@ -385,8 +385,12 @@ PdfDocument* PdfDocument::fromData(const QByteArray &data, QObject *parent)
     std::unique_ptr<PdfDocument> doc(new PdfDocument(parent));
     doc->d->m_pdfData = data;
     // PDFDoc takes ownership of stream
+#if KPOPPLER_VERSION < QT_VERSION_CHECK(26, 1, 90)
     auto stream = new MemStream(const_cast<char*>(doc->d->m_pdfData.constData()), 0, doc->d->m_pdfData.size(), Object());
-    std::unique_ptr<PDFDoc> popplerDoc(new PDFDoc(stream));
+#else
+    auto stream = std::make_unique<MemStream>(const_cast<char*>(doc->d->m_pdfData.constData()), 0, doc->d->m_pdfData.size(), Object());
+#endif
+    auto popplerDoc = std::make_unique<PDFDoc>(std::move(stream));
     if (!popplerDoc->isOk()) {
         qCWarning(Log) << "Got invalid PDF document!" << popplerDoc->getErrorCode();
         return nullptr;
-- 
GitLab

