diff -Npur more_itertools-11.1.0.orig/pyproject.toml more_itertools-11.1.0/pyproject.toml
--- more_itertools-11.1.0.orig/pyproject.toml	2025-11-04 23:49:26.103295300 +0100
+++ more_itertools-11.1.0/pyproject.toml	2026-09-04 05:51:02.454055484 +0200
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["flit_core >=3.12,<4"]
+requires = ["flit_core >=3.12,<5"]
 build-backend = "flit_core.buildapi"
 
 [project]
