From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 23 Sep 2025 16:20:50 -0500
Subject: [PATCH] Allow xdg directory to be overridden

---
 justfile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/justfile b/justfile
index 8fc71d1..7ebd8ba 100644
--- a/justfile
+++ b/justfile
@@ -16,7 +16,8 @@ export DISABLE_IF_EXISTS := disable-if-exists || env('DISABLE_IF_EXISTS', '')
 base-dir := absolute_path(clean(rootdir / prefix))
 cargo-target-dir := env('CARGO_TARGET_DIR', 'target')
 desktop-entry := appid + '.desktop'
-autostart-dst := rootdir / 'etc' / 'xdg' / 'autostart' / desktop-entry
+xdg-dir := env('XDG_INSTALL_DIR', 'etc' / 'xdg')
+autostart-dst := rootdir / xdg-dir / 'autostart' / desktop-entry
 autostart-entry := appid + '.Autostart.desktop'
 bin-dst := base-dir / 'bin' / name
 desktop-dst := base-dir / 'share' / 'applications' / desktop-entry
