From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Kope=C4=87?= <michal@nozomi.space>
Date: Sun, 15 Feb 2026 01:03:21 +0100
Subject: [PATCH 020/109] dcn316/dcn316_resource.c: instantiate HPO HDMI
 encoders

---
 .../dc/resource/dcn316/dcn316_resource.c      | 83 +++++++++++++++++++
 1 file changed, 83 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/dc/resource/dcn316/dcn316_resource.c b/drivers/gpu/drm/amd/display/dc/resource/dcn316/dcn316_resource.c
index 045ce01bd74e..d1962dd8a9cd 100644
--- a/drivers/gpu/drm/amd/display/dc/resource/dcn316/dcn316_resource.c
+++ b/drivers/gpu/drm/amd/display/dc/resource/dcn316/dcn316_resource.c
@@ -55,6 +55,8 @@
 #include "dcn30/dcn30_dio_stream_encoder.h"
 #include "dcn31/dcn31_hpo_dp_stream_encoder.h"
 #include "dcn31/dcn31_hpo_dp_link_encoder.h"
+#include "dcn30/dcn30_hpo_hdmi_stream_encoder.h"
+#include "dcn30/dcn30_hpo_hdmi_link_encoder.h"
 #include "dcn31/dcn31_apg.h"
 #include "dcn31/dcn31_dio_link_encoder.h"
 #include "dcn31/dcn31_vpg.h"
@@ -442,6 +444,17 @@ static const struct dcn31_hpo_dp_stream_encoder_mask hpo_dp_se_mask = {
 	DCN3_1_HPO_DP_STREAM_ENC_MASK_SH_LIST(_MASK)
 };
 
+static struct dcn30_hpo_hdmi_stream_encoder_registers hpo_hdmi_stream_enc_regs = {
+	DCN3_0_HPO_HDMI_STREAM_ENC_REG_LIST()
+};
+
+static const struct dcn30_hpo_hdmi_stream_encoder_shift hpo_hdmi_se_shift = {
+	DCN3_0_HPO_HDMI_STREAM_ENC_MASK_SH_LIST(__SHIFT)
+};
+
+static const struct dcn30_hpo_hdmi_stream_encoder_mask hpo_hdmi_se_mask = {
+	DCN3_0_HPO_HDMI_STREAM_ENC_MASK_SH_LIST(_MASK)
+};
 
 #define hpo_dp_link_encoder_reg_list(id)\
 [id] = {\
@@ -466,6 +479,17 @@ static const struct dcn31_hpo_dp_link_encoder_mask hpo_dp_le_mask = {
 	DCN3_1_HPO_DP_LINK_ENC_MASK_SH_LIST(_MASK)
 };
 
+static struct dcn30_hpo_hdmi_link_encoder_registers hpo_hdmi_link_enc_regs = {
+	DCN3_0_HPO_HDMI_LINK_ENC_REG_LIST()
+};
+
+static const struct dcn30_hpo_hdmi_link_encoder_shift hpo_hdmi_le_shift = {
+	DCN3_0_HPO_HDMI_LINK_ENC_MASK_SH_LIST(__SHIFT)
+};
+
+static const struct dcn30_hpo_hdmi_link_encoder_mask hpo_hdmi_le_mask = {
+	DCN3_0_HPO_HDMI_LINK_ENC_MASK_SH_LIST(_MASK)
+};
 
 #define dpp_regs(id)\
 [id] = {\
@@ -813,6 +837,8 @@ static const struct resource_caps res_cap_dcn31 = {
 	.num_dig_link_enc = 5,
 	.num_hpo_dp_stream_encoder = 4,
 	.num_hpo_dp_link_encoder = 2,
+	.num_hpo_hdmi_stream_encoder = 1,
+	.num_hpo_hdmi_link_encoder = 1,
 	.num_pll = 5,
 	.num_dwb = 1,
 	.num_ddc = 5,
@@ -1297,6 +1323,44 @@ static struct hpo_dp_stream_encoder *dcn31_hpo_dp_stream_encoder_create(
 	return &hpo_dp_enc31->base;
 }
 
+static struct hpo_hdmi_stream_encoder *dcn316_hpo_hdmi_stream_encoder_create(
+	enum engine_id eng_id,
+	struct dc_context *ctx)
+{
+	struct dcn30_hpo_hdmi_stream_encoder *hpo_hdmi_enc30;
+	struct vpg *vpg;
+	struct afmt *afmt;
+	uint32_t hpo_hdmi_inst;
+	uint32_t vpg_inst;
+	uint32_t afmt_inst;
+
+	hpo_hdmi_inst = eng_id - ENGINE_ID_HPO_0;
+
+	/* Mapping of VPG register blocks to HPO HDMI block instance (guessed):
+	 * VPG[5] -> HPO_HDMI[0]
+	 * AFMT[5] -> HPO_HDMI[0]
+	 */
+	vpg_inst = hpo_hdmi_inst + 5;
+	afmt_inst = hpo_hdmi_inst + 5;
+
+	/* allocate HPO stream encoder and create VPG sub-block */
+	hpo_hdmi_enc30 = kzalloc(sizeof(struct dcn30_hpo_hdmi_stream_encoder), GFP_KERNEL);
+	vpg = dcn31_vpg_create(ctx, vpg_inst);
+	afmt = dcn31_afmt_create(ctx, afmt_inst);
+
+	if (!hpo_hdmi_enc30 || !vpg || !afmt) {
+		kfree(hpo_hdmi_enc30);
+		kfree(vpg);
+		return NULL;
+	}
+
+	dcn30_hpo_hdmi_stream_encoder_construct(hpo_hdmi_enc30, ctx, hpo_hdmi_inst,
+					vpg, afmt, &hpo_hdmi_stream_enc_regs,
+					&hpo_hdmi_se_shift, &hpo_hdmi_se_mask);
+
+	return &hpo_hdmi_enc30->base;
+}
+
 static struct hpo_dp_link_encoder *dcn31_hpo_dp_link_encoder_create(
 	uint8_t inst,
 	struct dc_context *ctx)
@@ -1315,6 +1379,23 @@ static struct hpo_dp_link_encoder *dcn31_hpo_dp_link_encoder_create(
 	return &hpo_dp_enc31->base;
 }
 
+static struct hpo_hdmi_link_encoder *dcn316_hpo_hdmi_link_encoder_create(
+	uint8_t inst,
+	struct dc_context *ctx)
+{
+	struct dcn30_hpo_hdmi_link_encoder *hpo_hdmi_enc30;
+
+	/* allocate HPO link encoder */
+	hpo_hdmi_enc30 = kzalloc(sizeof(struct dcn30_hpo_hdmi_link_encoder), GFP_KERNEL);
+	if (!hpo_hdmi_enc30)
+		return NULL; /* out of memory */
+
+	hpo_hdmi_link_encoder31_construct(hpo_hdmi_enc30, ctx, inst,
+					&hpo_hdmi_link_enc_regs,
+					&hpo_hdmi_le_shift, &hpo_hdmi_le_mask);
+
+	return &hpo_hdmi_enc30->base;
+}
 
 static struct dce_hwseq *dcn31_hwseq_create(
 	struct dc_context *ctx)
@@ -1335,6 +1416,8 @@ static const struct resource_create_funcs res_create_funcs = {
 	.create_stream_encoder = dcn316_stream_encoder_create,
 	.create_hpo_dp_stream_encoder = dcn31_hpo_dp_stream_encoder_create,
 	.create_hpo_dp_link_encoder = dcn31_hpo_dp_link_encoder_create,
+	.create_hpo_hdmi_stream_encoder = dcn316_hpo_hdmi_stream_encoder_create,
+	.create_hpo_hdmi_link_encoder = dcn316_hpo_hdmi_link_encoder_create,
 	.create_hwseq = dcn31_hwseq_create,
 };
 
