--- a/steamos-manager/src/platform.rs	2026-09-11 09:04:21.259657156 +0200
+++ b/steamos-manager/src/platform.rs	2026-09-11 09:11:00.359253227 +0200
@@ -38,8 +38,8 @@ pub(crate) struct SessionConfig {
 impl Default for SessionConfig {
     fn default() -> Self {
         SessionConfig {
-            gamescope_session_service: String::from("gamescope-session-plus@ogui-steam.service"),
-            gamescope_session_desktop: String::from("gamescope-session-ogui-steam.desktop"),
+            gamescope_session_service: String::from("gamescope-session-plus@steam-plus.service"),
+            gamescope_session_desktop: String::from("gamescope-session-steam-plus.desktop"),
             desktop: String::from("plasma.desktop"),
         }
     }
