From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 22 Apr 2026 15:52:14 -0500
Subject: [PATCH] Try versioned zig libdir

---
 src/introspect.zig | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/introspect.zig b/src/introspect.zig
index 8467b566c6..3164bc492d 100644
--- a/src/introspect.zig
+++ b/src/introspect.zig
@@ -15,6 +15,19 @@ const build_options = @import("build_options");
 fn testZigInstallPrefix(base_dir: fs.Dir) ?Cache.Directory {
     const test_index_file = "std" ++ fs.path.sep_str ++ "std.zig";
 
+    // Try the versioned libdir first
+    zig_dir_versioned: {
+        // Try lib/zig-0.15/std/std.zig
+        const lib_zig = "lib" ++ fs.path.sep_str ++ "zig-0.15";
+        var test_zig_dir = base_dir.openDir(lib_zig, .{}) catch break :zig_dir_versioned;
+        const file = test_zig_dir.openFile(test_index_file, .{}) catch {
+            test_zig_dir.close();
+            break :zig_dir_versioned;
+        };
+        file.close();
+        return .{ .handle = test_zig_dir, .path = lib_zig };
+    }
+
     zig_dir: {
         // Try lib/zig/std/std.zig
         const lib_zig = "lib" ++ fs.path.sep_str ++ "zig";
